kafka/arena-acl-ingestor/src/main/kotlin/no/nav/amt/tiltak/ingestors/arena_acl_ingestor/processor/DeltakerProcessor.kt

```diff
@@ -42,40 +44,52 @@ class DeltakerProcessor(
 	private fun upsert(message: MessageWrapper<DeltakerPayload>) {
 		val deltakerDto = message.payload
 
-		val gjennomforingId = gjennomforingService.getGjennomforingOrNull(deltakerDto.gjennomforingId)?.id
-			?: upsertGjennomforing(deltakerDto.gjennomforingId).id
+		val (gjennomforingId, tiltakstype) = getGjennomforingIdAndTiltakstype(deltakerDto.gjennomforingId)
 
-		val status = DeltakerStatusInsert(
-			id = UUID.randomUUID(),
-			deltakerId = deltakerDto.id,
-			type = tilDeltakerStatusType(deltakerDto.status),
-			aarsak = tilDeltakerAarsak(deltakerDto.statusAarsak),
-			aarsaksbeskrivelse = null,
-			gyldigFra = deltakerDto.statusEndretDato,
-		)
+		val erKometDeltaker = unleashService.erKometMasterForTiltakstype(tiltakstype)
 
-		val deltakerUpsert = DeltakerUpsert(
-			id = deltakerDto.id,
-			statusInsert = status,
-			startDato = deltakerDto.startDato,
-			sluttDato = deltakerDto.sluttDato,
-			dagerPerUke = deltakerDto.dagerPerUke,
-			prosentStilling = deltakerDto.prosentDeltid,
-			registrertDato = deltakerDto.registrertDato,
-			gjennomforingId = gjennomforingId,
-			innsokBegrunnelse = deltakerDto.innsokBegrunnelse,
-			innhold = null,
-			kilde = Kilde.ARENA,
-			forsteVedtakFattet = null,
-			sistEndretAv = null,
-			sistEndretAvEnhet = null
-		)
+		if (!erKometDeltaker) {
+			val status = DeltakerStatusInsert(
+				id = UUID.randomUUID(),
+				deltakerId = deltakerDto.id,
+				type = tilDeltakerStatusType(deltakerDto.status),
+				aarsak = tilDeltakerAarsak(deltakerDto.statusAarsak),
+				aarsaksbeskrivelse = null,
+				gyldigFra = deltakerDto.statusEndretDato,
+			)
 
-		transactionTemplate.executeWithoutResult {
-			deltakerService.upsertDeltaker(deltakerDto.personIdent, deltakerUpsert)
+			val deltakerUpsert = DeltakerUpsert(
+				id = deltakerDto.id,
+				statusInsert = status,
+				startDato = deltakerDto.startDato,
+				sluttDato = deltakerDto.sluttDato,
+				dagerPerUke = deltakerDto.dagerPerUke,
+				prosentStilling = deltakerDto.prosentDeltid,
+				registrertDato = deltakerDto.registrertDato,
+				gjennomforingId = gjennomforingId,
+				innsokBegrunnelse = deltakerDto.innsokBegrunnelse,
+				innhold = null,
+				kilde = Kilde.ARENA,
+				forsteVedtakFattet = null,
+				sistEndretAv = null,
+				sistEndretAvEnhet = null
+			)
+
+			transactionTemplate.executeWithoutResult {
+				deltakerService.upsertDeltaker(deltakerDto.personIdent, deltakerUpsert, erKometDeltaker)
```

Comment:
Stusser på denne. Vil vi ikke beholde kilden til en arenadeltaker om man får ny status fra arena?

Reply:
Hvis komet har tatt over deltakeren skal vi aldri motta oppdatering fra arena?

Reply:
Ja sant, ble nok bare forvirret av at man på en måte "kalkulerte" om det er en komet deltaker eller ikke(vha tiltakstype) og setter dette isteden for å la den faktiske kilden til deltakeren bestemme. 
Jeg ser nå at upsertDeltaker bruker denne egentlig til, ikke publiser deltaker på topicen, så da ga det menig

(Hvis jeg kan "tenke litt høyt")
I praksis så blir det vel kanskje det samme, så lenge timingen blir riktig med at vi sørger for at alle arenaendringer på topicen er ferdig prosessert før vi skrur på toggle. 

Hvis vi relaster data på topicen etter at toggelen er på så blir de ignorert, og det virker riktig.

Reply:
Vi har diskutert dette på slack :)